.github/workflows/ci.yml

```diff
@@ -1089,6 +1089,7 @@ jobs:
           -DDOWNLOAD_EIGEN=ON
           -DCMAKE_CXX_COMPILER=clang++
           -DCMAKE_CXX_STANDARD=17
+          -DPYBIND11_PYTEST_ARGS='-k "not test_throw_nested_exception"'
```

Comment:
Drive-by comment/question:

This introduces a new way to skip tests and might be very surprising. I can see myself being very confused assuming the test works on all platforms until I finally discover this line here.

I feel it would be better to use `@pytest.mark.xfail(...)`, similar to what we have already in test_exceptions.py:

* https://github.com/pybind/pybind11/blob/8b48ff878c168b51fe5ef7b8c728815b9e1a9857/tests/test_exceptions.py#L74-L79

* https://github.com/pybind/pybind11/blob/8b48ff878c168b51fe5ef7b8c728815b9e1a9857/tests/test_exceptions.py#L100-L102

What do you think?

Reply:
I think it has to be an skipif, as xfail will still try to run it, and it can't recover a segfault. I didn't know we had a way to query the compiler info, I can use that. I don't know how specific this is, 3.10 only or all versions.

Reply:
> I think it has to be an skipif, as xfail will still try to run it, and it can't recover a segfault.

Good point.

>  I didn't know we had a way to query the compiler info

It's a bit ad-hoc. In this case I see this in the pytest output:

```
C++ Info: MSVC 193933522 C++17 __pybind11_internals_v5_msvc_mscver1939_debug__ PYBIND11_SIMPLE_GIL_MANAGEMENT=False PYBIND11_NUMPY_1_ONLY=False
```

While this is what cmake reports:

```
-- The CXX compiler identification is Clang 18.1.0 with GNU-like command-line
```

Having access to the clang version via some attribute of `pybind11_tests` would seem ideal. I think it would be easy to add around here:

* https://github.com/pybind/pybind11/blob/8b48ff878c168b51fe5ef7b8c728815b9e1a9857/tests/pybind11_tests.cpp#L90

What do you think? (I could help with that later tonight if you agree with the idea.)

Reply:
pybind11_tests.compiler_info should in theory work? I've already started it anyway. :)

Reply:
It does if you flip the order.